math/src/main/java/com/powsybl/math/matrix/DenseMatrix.java

```diff
@@ -260,6 +260,18 @@ protected int getEstimatedNonZeroValueCount() {
         return getRowCount() * getColumnCount();
     }
 
+    @Override
+    public DenseMatrix transpose() {
+        double[] transposed = new double[columnCount * rowCount];
+        for (int i = 0; i < rowCount; i++) {
+            for (int j = 0; j < columnCount; j++) {
+                double value = buffer.getDouble(j * Double.BYTES * rowCount + i * Double.BYTES);
+                transposed[i * columnCount + j] = value;
+            }
+        }
+        return new DenseMatrix(columnCount, rowCount, transposed);
```

Comment:
ouch, not an implementation meant to be very effective, but we have here a `columnCount * rowCount` temporary array. If we use the `set(int, int, double)`, we avoid that, at the cost of the `checkBounds` for each though.

Reply:
Right, fixed by directly filling a bytebuffer, we can avoid unnecessary bounds check.